### PR TITLE
Fix #72

### LINKED ISSUE #72: [codegen] conversion from FREE_IMAGE_FORMAT to itself is pointless
same input example as https://github.com/genotrance/nimterop/issues/71

Hint: conversion from FREE_IMAGE_FORMAT to itself is pointless [ConvFromXtoItselfNotNeeded]

comes from this:
```nim
type FREE_IMAGE_FORMAT* = distinct int
converter enumToInt(en: FREE_IMAGE_FORMAT): int {.used.} = en.int
```

### PR DESCRIPTION
**Pull request by genotrance**

### CODE REVIEW

File: config.nims [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+scanner.cc, which would otherwise result in link errors such as:
 "std::terminate()", referenced from:
       ___clang_call_terminate in scanner.cc.o
 ]#
 when defined(MacOSX):
   switch("clang.linkerexe", "g++")
 else:
   switch("gcc.linkerexe", "g++")
+
+# Workaround for NilAccessError crash on Windows
```
Comment (timotheecour) [inline config.nims]: link to https://github.com/genotrance/nimterop/issues/98

File: nimterop/globals.nim [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import sequtils, sets, tables
+import segfaults, sequtils, sets, tables
```
Comment (timotheecour) [inline nimterop/globals.nim]: add: pending https://github.com/genotrance/nimterop/issues/98 ; this import has implications (and its own issues)

File: nimterop/ast.nim [hunk 3]
```diff
@@ -134,6 +134,9 @@ proc printNimHeader*() =
 #   $2 $3
 
 {.experimental: "codeReordering".}
+{.hint[ConvFromXtoItselfNotNeeded]: off.}
```
Comment (timotheecour) [inline nimterop/ast.nim]: this is hiding issue under a rug though... we should at least keep an issue open to track this or at least a TODO in code

File: nimterop/types.nim [hunk 4]
```diff
@@ -31,3 +31,31 @@ when defined(c):
 elif defined(cpp):
   type
     wchar_t* {.importc.} = object
+
+template enumOp*(op, typ, typout) =
+  proc op*(x: typ, y: int): typout {.borrow.}
+  proc op*(x: int, y: typ): typout {.borrow.}
+  proc op*(x, y: typ): typout {.borrow.}
+
+template defineEnum*(typ: untyped) =
```
Comment (timotheecour) [inline nimterop/types.nim]: this should be opt-in for type safety

Comment (timotheecour) [inline nimterop/types.nim]: can we use `s/untyped/typed/` ? (always prefer the least powerful idiom)

File: nimterop/types.nim [hunk 5]
```diff
@@ -31,3 +31,31 @@ when defined(c):
...
+
+  proc `shl`*(x: typ, y: int): typ {.borrow.}
+  proc `shl`*(x: int, y: typ): typ {.borrow.}
+  proc `shl`*(x, y: typ): typ {.borrow.}
+
+  proc `shr`*(x: typ, y: int): typ {.borrow.}
+  proc `shr`*(x: int, y: typ): typ {.borrow.}
+  proc `shr`*(x, y: typ): typ {.borrow.}
+
+  proc `$` *(x: typ): string {.borrow.}
```
Comment (timotheecour) [inline nimterop/types.nim]: can't we instead use enum element name in source code? treesitter shd expose it, eg:
`$blue => "blue" instead of 23`

**Comment by timotheecour** [comment]
for the mapping of enums in this PR I'd really switch to this: https://github.com/genotrance/nimterop/issues/99

File: nimterop/globals.nim (hunk 2 above)
Comment (genotrance) [reply nimterop/globals.nim]: Can you explain? Seems nice to get a stack trace when things crash.

Comment (timotheecour) [reply nimterop/globals.nim]: so it's indeed nice when debugging, but i don't think it should be enabled by default as it resets signal handler (eg which a user could've set etc)

File: nimterop/types.nim (hunk 4 above)
Comment (genotrance) [reply nimterop/types.nim]: `typ: enum` doesn't work - Error: undeclared identifier: 'ENUM', neither does `typ: typed`. Am just removing it altogether for now.

**Pull request merged**